src/ctapipe/monitoring/calculator.py

```diff
@@ -106,27 +105,30 @@ def __init__(
                 name, subarray=self.subarray, parent=self
             )
 
-        # Initialize the instances of OutlierDetector
-        self.outlier_detectors = {}
+        # Initialize the instances of OutlierDetector from the configuration
+        self.outlier_detectors, self.apply_to_list = [], []
         if self.outlier_detector_list is not None:
             for d, outlier_detector in enumerate(self.outlier_detector_list):
+                outlier_detector = outlier_detector.copy()
```

Comment:
Since you now introduced the `config` key and are not actually passing the `**outlier_detector` dict itself to the constructor, the copy and pop are not necessary.

Simply iterating over the dicts and accessing the three keys is fine